### PR TITLE
GetClient return error info

### PR DESCRIPTION
**Pull request by kadisi**
Signed-off-by: zhangjie <iamkadisi@163.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

/kind feature
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #
GetClient return detail error info. not only output websocket error.

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by m1093782566** [comment]
/lgtm

leave approve to @sids-b @edisonxiang

File: edge/pkg/edgehub/clients/factory.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package clients
 
 import (
+	"fmt"
```
Comment (sids-b) [inline edge/pkg/edgehub/clients/factory.go]: nit: please seperate golang and kubeedge packages with a blank line

**Comment by sids-b** [comment]
I think we need to fmt the file and fix the UT

File: edge/pkg/edgehub/clients/factory.go [hunk 2]
```diff
@@ -13,10 +15,14 @@ const (
 	ClientTypeQuic      = "quic"
 )
 
+// ErrorWrongType is Wrong Type Error
+var ErrorWrongType = errors.New("Wrong Client Type")
```
Comment (m1093782566) [inline edge/pkg/edgehub/clients/factory.go]: nit: s/ErrorWrongType /ErrorWrongClientType

File: edge/pkg/edgehub/clients/factory.go [hunk 3]
```diff
@@ -39,11 +45,10 @@ func GetClient(clientType string, config *config.EdgeHubConfig) Adapter {
 			ProjectID:        config.CtrConfig.ProjectID,
 			NodeID:           config.CtrConfig.NodeID,
 		}
-		return quicclient.NewQuicClient(&quicConfig)
-	} else {
-		log.LOGGER.Errorf("donot support client type: %s", clientType)
+		return quicclient.NewQuicClient(&quicConfig), nil
+	default:
+		log.LOGGER.Errorf("Do not support client type: %s", clientType)
```
Comment (m1093782566) [inline edge/pkg/edgehub/clients/factory.go]: What about the error message,:

"Client type: %s is not supported"

Comment (sids-b) [reply edge/pkg/edgehub/clients/factory.go]: +1

**Comment by sids-b** [comment]
@kadisi ,Would you like to join KubeEdge as a member ?

**Comment by sids-b** [comment]
You can find information about KubeEdge community membership roles in this document https://github.com/kubeedge/kubeedge/blob/master/docs/getting-started/community-membership.md

**Comment by kadisi** [comment]
@sids-b thank you,  I really want to contribute to kubeedge and join kubeedge as a member.

**Comment by sids-b** [comment]
> @sids-b thank you, I really want to contribute to kubeedge and join kubeedge as a member.

We are very glad to have contributors like you. I have sent you the invitation :)

File: edge/pkg/edgehub/clients/factory.go [hunk 4]
```diff
@@ -13,10 +15,14 @@ const (
 	ClientTypeQuic      = "quic"
 )
 
+// ErrorWrongClientType is Wrong Client Type Error
+var ErrorWrongClientType = errors.New("Wrong Client Type")
```
Comment (m1093782566) [inline edge/pkg/edgehub/clients/factory.go]: Sorry for nit picking, we are following the convension that error message is started with lower-case word while log message is started with uppper-case word. So in this case, the error string should be "wrong Client Type".

Comment (kadisi) [reply edge/pkg/edgehub/clients/factory.go]: ok

File: edge/pkg/edgehub/clients/factory_test.go [hunk 5]
```diff
@@ -59,15 +60,16 @@ func TestGetClient(t *testing.T) {
...
 
 		{"TestGetClient: Negative Test Case", args{
 			clientType: "WrongClientType",
-		}, nil},
+		}, nil, ErrorWrongClientType},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			if got := GetClient(tt.args.clientType, tt.args.config); !reflect.DeepEqual(got, tt.want) {
+			if got, err := GetClient(tt.args.clientType, tt.args.config); !reflect.DeepEqual(got, tt.want) || err != tt.err {
```
Comment (m1093782566) [inline edge/pkg/edgehub/clients/factory_test.go]: I think we can split the error checks but I don't want to block this PR too long. We can improve it in follow-ups.

Comment (kadisi) [reply edge/pkg/edgehub/clients/factory_test.go]: ok

**Pull request merged**